pkg/controller/controller.go

```diff
@@ -84,15 +86,17 @@ type ControllerParams struct {
 //   in a way that will delay destruction throughout the controller run or to
 //   check for the destruction throughout the run.
 type Controller struct {
-	name           string
-	params         ControllerParams
-	successCount   int
-	failureCount   int
-	lastError      error
-	lastErrorStamp time.Time
-	uuid           string
-	stop           chan struct{}
-	mutex          lock.RWMutex
+	name              string
+	params            ControllerParams
+	successCount      int
+	lastSuccessStamp  time.Time
+	failureCount      int
+	consecutiveErrors int
+	lastError         error
+	lastErrorStamp    time.Time
+	uuid              string
+	stop              chan struct{}
+	mutex             lock.RWMutex
```

Comment:
In other structures we have put the mutex above all of the resources which should be accessed under that lock - can you do that here?

Reply:
fixed